workers.py

```diff
@@ -44,6 +44,100 @@ def _retryable(method: str, *args, **kwargs) -> requests.Response:
     raise requests.HTTPError('All attempts failed')
 
 
+def query_main_collection(
+        host: str,
+        endpoint: str,
+        main_collection: str,
+        query_string: str
+) -> dict:
+    """Query a Collection.
+
+    Returns a Response object with data retrieved from all pages
+    :param host: The host
+    :param endpoint: API Endpoint
+    :param main_collection: Collection name ('container_nodes', 'sources' etc)
+    :param query_string: Query string as additional params to the GET request
+    :return: Response object with data retrieved from all pages
+    :raises: HTTPError in the function caller
+    """
+    # First GET call to get data as well as pagination links
+    resp = _retryable(
+        'get',
+        f'{host}{endpoint}/{main_collection}',
+        params={query_string: ''},
+        verify=False
+    )
+    out = resp.json()
+    all_data = out['data']
+    prometheus_metrics.METRICS['get_successes'].inc()
+
+    # Subsequent GET calls that reference the pagination link
+    while out['links'].get('next'):
+        resp = _retryable(
+            'get',
+            f'{host}{out["links"]["next"]}',
+            verify=False
+        )
+        out = resp.json()
+        all_data += out['data']
+        prometheus_metrics.METRICS['get_successes'].inc()
+    return all_data
+
+
+def query_sub_collection(
+        host: str,
+        endpoint: str,
+        main_collection: str,
+        sub_collection: str,
+        query_string: str,
+        collection: list,
+        foreign_key: str
+) -> dict:
+    """Query a SubCollection for all records in the main collection.
+
+    Returns a Response object with data retrieved from all pages
+    :param host: The host
+    :param endpoint: API Endpoint
+    :param main_collection: Collection name ('container_nodes', 'sources' etc)
+    :param sub_collection: Sub-Collection name ('tags')
+    :param query_string: Query string as additional params to the GET request
+    :param collection: List of all main collection items
+    :param foreign_key: Foreign key to be added to sub-collection records
+    :return: Response object with sub-collection data retrieved from all pages
+    :raises: HTTPError in the function caller
+    """
+    all_data = []
+    for item in collection:
+        # First GET call to get data as well as pagination links
+        resp = _retryable(
+            'get',
+            f'{host}{endpoint}'
+            f'/{main_collection}/{item["id"]}/{sub_collection}',
+            params={query_string: ''},
+            verify=False
+        )
+        out = resp.json()
+
+        for row in enumerate(out['data']):
```

Comment:
Why you do enumerate instead of `for row in out['data']`? You're not using the index, don't you? So what's the purpose.

Enumerate would convert something like `['value']` to `[(0, 'value'),]`. And since you intent to use the `'value'` only, why do you need the enumerate?

Reply:
That is correct -- no need for enumerate here